Ada/videos (#23)
Botcopy/docs: docs/advanced/handover.md

Here are the steps that proxy should follow:

1. You receive a webhook call when Botcopy detects the context/session parameter of `bcHumanHandover`, your bot is paused and the end-user sends a message.
2. You can interact with the end-user through `message`, `channel_update`, and `agent_typing`.
3. You will handle the socket towards the chosen ITSM tool.

## Setup in the Botcopy Portal

> The handover integration is set for each individual Botcopy bot on its Connect page in the Portal. You may set the webhook URL, a context/session parameter of `bcHumanHandover` which will trigger the handover, and enable the storing of message history. This is also where you generate your `organization access token` to view your message history, and set a `bot access token` so the requests can authenticate with your webhook.

Reviewer: we are force to use `bcHumanHandover` for CX bots due to a bug that was fixed specifically for this session parameter value on the chat widget.
But using `bcHumanHandover` will cause a bug with Dialogflow ES, output context should be lowercase so we could write this instead.
```suggestion
The handover integration is set for each individual Botcopy bot on its Connect page in the Portal.
You may set your Live Chat Endpoint Webhook URL
If you're using Dialogflow ES set any lowercased trigger context you want like `bc-human-handover`
If you're using Dialoflow CX set `bcHumanHandover` as your trigger session parameter
Enable the storing of message history.
Generate your `organization access token`, that's used by Botcopy to authenticate you when you make a POST request to `https://api.botcopy.com/webhooks/handover/push`
Set a `bot access token` that you will use to authenticate Botcopy when Botcopy is making POST requests to your Live Chat Endpoint Webhook.
```

**Direction:** Botcopy -> Client

> **Purpose:** Client receives a webhook call when Botcopy detects a specific context in a Dialogflow response. Client defines a webhook URL pointing to your middleware, a bot access token to authenticate with your middleware, and a Dialogflow context name to trigger the webhook. When the context is detected, we send a `POST` request to your webhook with the following JSON body:

Reviewer: ```suggestion
**Purpose:** Client receives a POST request to their Live Chat Endpoint Webhook when Botcopy detects a trigger output context(ES)/session parameter(CX) in a Dialogflow response.
```

}
```

Client must respond with a JSON body `{ paused: true, minutesPaused: number}` within 5 seconds, otherwise the bot will not pause the conversation and ignore the response after the timeout. if `{paused: false}` is provided, the bot will not pause the conversation.

> If the JSON body is received, the conversation is paused and messages from the chat user are sent to your middleware instead of Dialogflow.

Reviewer: ```suggestion
```

> If `minutesPaused` is not provided, the default is set to ten minutes.

Reviewer: ```suggestion
If `minutesPaused` is not provided, the default is set to 10 minutes.
```

> When sending each message to the user, paused and minutesPaused should be a part of the payload that you send to Botcopy's endpoint: `https://api.botcopy.com/webhooks/handover/push`. To keep the conversation going make sure that `paused` is true and `minutesPaused` greater than 0. This way the session time expiry is pushed back each time a message is received from and sent to the end-user and you have a way to resume AI responses on your bot when you receive or send a message to the user.

Reviewer: ```suggestion
When sending each message to the user, `paused` and `minutesPaused` should be a part of the payload that you send to Botcopy's endpoint: `https://api.botcopy.com/webhooks/handover/push`. To keep the conversation going make sure that `paused` is true and `minutesPaused` greater than 0. This way the session time expiry is pushed back each time a message is received from and sent to the end-user and you have a way to resume AI responses on your bot when you receive or send a message to the user.
```

> You can reference the specific Dialogflow session with the `session` key of the initial request.

Reviewer: ```suggestion
```

```json5
{
  accessToken: "***", // bot access token
  userId: "***",
  botId: "***",

>   session: "projects/test***/agent/sessions/***-***-***-***-***",

Reviewer: ```suggestion
```

}
```

> Client must respond JSON body that includes `{ paused: true, minutesPaused: number }` within 5 seconds to continue pausing the bot. `minutesPaused` defaults to 10 minutes.

Reviewer: It's already above
```suggestion
```
